core/src/main/java/eu/fasten/core/maven/data/Dependency.java

```diff
@@ -15,235 +15,329 @@
+        hashCode = prime * hashCode + ((artifactId == null) ? 0 : artifactId.hashCode());
+        hashCode = prime * hashCode + (optional ? 1231 : 1237);
+
+        hashCode = prime * hashCode + ((classifier == null) ? 0 : classifier.hashCode());
+        hashCode = prime * hashCode + ((scope == null) ? 0 : scope.hashCode());
+        hashCode = prime * hashCode + ((type == null) ? 0 : type.hashCode());
+
+        hashCode = prime * hashCode + ((versionConstraint == null) ? 0 : versionConstraint.hashCode());
+        hashCode = prime * hashCode + ((versionConstraints == null) ? 0 : versionConstraints.hashCode());
+        hashCode = prime * hashCode + ((exclusions == null) ? 0 : exclusions.hashCode());
+        return hashCode;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (getClass() != obj.getClass())
+            return false;
+        Dependency other = (Dependency) obj;
+        if (hashCode != other.hashCode) {
+            return false;
+        }
+        if (artifactId == null) {
+            if (other.artifactId != null)
+                return false;
+        } else if (!artifactId.equals(other.artifactId))
+            return false;
+        if (classifier == null) {
+            if (other.classifier != null)
+                return false;
+        } else if (!classifier.equals(other.classifier))
+            return false;
+        if (exclusions == null) {
+            if (other.exclusions != null)
+                return false;
+        } else if (!exclusions.equals(other.exclusions))
+            return false;
+        if (ga == null) {
+            if (other.ga != null)
+                return false;
+        } else if (!ga.equals(other.ga))
+            return false;
+        if (groupId == null) {
+            if (other.groupId != null)
+                return false;
+        } else if (!groupId.equals(other.groupId))
+            return false;
+        if (optional != other.optional)
+            return false;
+        if (scope != other.scope)
+            return false;
+        if (type == null) {
+            if (other.type != null)
+                return false;
+        } else if (!type.equals(other.type))
+            return false;
+        if (versionConstraint == null) {
+            if (other.versionConstraint != null)
+                return false;
+        } else if (!versionConstraint.equals(other.versionConstraint))
+            return false;
+        if (versionConstraints == null) {
+            if (other.versionConstraints != null)
+                return false;
+        } else if (!versionConstraints.equals(other.versionConstraints))
+            return false;
+        return true;
+    }
+
+    @Override
+    public String toString() {
+        return ToStringBuilder.reflectionToString(this, ToStringStyle.MULTI_LINE_STYLE);
+    }
+
+    // TODO delete everything below
```

Comment:
Do you still want to do this TODO?

Reply:
This is just a TODO for the future. At some point, the remaining code in that class should be removed by refactoring the JSON handling. This is out-of-scope for this PR though.